airflow.cfg

```diff
@@ -972,7 +972,7 @@ scheduler_idle_sleep_time = 1
 # Number of seconds after which a DAG file is parsed. The DAG file is parsed every
 # ``min_file_process_interval`` number of seconds. Updates to DAGs are reflected after
 # this interval. Keeping this number low will increase CPU usage.
-min_file_process_interval = 30
+min_file_process_interval = 300
```

Comment:
I noticed that my log files were filling very quickly, and it was because Airflow re-parses every DAG file at this interval (regardless of if it's switched on/scheduled) and logs the result.

It seems like a lot of overhead considering our DAG code very rarely changes, but the obvious down side is that if your DAG's code changes, you have to wait. Frustratingly, there does not seem to be a way to manually trigger the DAGs to re-parse.

I'll keep looking into ways around this, but one option would be to set this to a higher number here (more than 30 seconds), but override the setting in the docker compose for local dev.

Reply:
also worth noting, there is a separate setting for how often airflow checks for brand new DAGS. that's called 
 `dag_dir_list_interval`  and the default setting is `300` seconds.

Reply:
sorry for all the noise here: i'm reading through [more docs](https://airflow.apache.org/docs/apache-airflow/stable/faq.html#faq) about DAG parsing and scheduler, and i want to test my assumptions some more!

Reply:
hopefully the last update from me on this. here's what i found after running some tests:
- `min_file_process_interval`: how frequently airflow re-processes DAG files. this is roughly the time it takes for modified DAG code changes to take effect. (not accounting for serialization: i'm not attempting to understand how that factors in). however, there's one caveat:

> The DAG parser will skip the `min_file_process_interval` check if a file is recently modified.

In my experience testing this, the `min_file_process_interval` was always skipped because changing a DAGs code means...modifying it. I think this behavior is not making much sense b/c it's intended to support use cases where you have thousands of DAGs 🤷 

TLDR: i think we should set the `min_file_process_interval` to at least 5 minutes because it will have no affect when making small + infrequent changes to DAGs, because those changes get picked up immediately.

- `dag_dir_list_interval` - this is the frequency with which Airflow will process brand new DAGs added to the `/dags` folder. we have it set to 5 minutes (the default). unfortunately, there is afaik no way to manually trigger Airflow to load in new DAGs, which means that 99.99% of the time there will be no new DAGs to be processed, and .01% of the time one of us will be waiting a few minutes to see our new DAG show up in the list. 

TLDR: I think we should we should set this to a shorter interval (60 seconds?) in local dev only.